lhotse/recipes/mtedx.py

```diff
@@ -0,0 +1,349 @@
+            f'http://www.openslr.org/resources/100/mtedx_{lang}.tgz',
+            filename=tar_path,
+            desc=f'Downloading MTEDx {lang}',
+        )
+        corpus_dir = target_dir / f'{lang}-{lang}.tgz'
+        completed_detector = target_dir / f'.{lang}.completed'
+        if not completed_detector.is_file():
+            shutil.rmtree(corpus_dir, ignore_errors=True)
+            with tarfile.open(tar_path) as tar:
+                tar.extractall(path=target_dir)
+                completed_detector.touch()
+
+
+###############################################################################
+#                              Prepare MTEDx                                
+###############################################################################
+def prepare_mtedx(
+    corpus_dir: Pathlike,
+    output_dir: Optional[Pathlike] = None,
+    languages: Optional[Union[str, Sequence[str]]] = 'all',
+    num_jobs: int = 1,
+) -> Dict[str, Dict[str, Dict[str, Union[RecordingSet, SupervisionSet]]]]:
+    """
+    Prepares manifest of all MTEDx languages requested.
+    
+    :param corpus_dir: Path to the root where MTEDx data was downloaded.
+                       It should be called mtedx_corpus.
+    :param output_dir: Root directory where .json manifests are stored.
+    :param languages: str or str sequence specifying the languages to prepare.
+        The str 'all' prepares all languages.
+    :return:
+    """
+    # Resolve corpus_dir type
+    if isinstance(corpus_dir, str):
+        corpus_dir = Path(corpus_dir)
+    
+    # Resolve output_dir type
+    if isinstance(output_dir, str):
+        output_dir = Path(output_dir)
+
+    langs_list = list(ISOCODE2LANG.keys())
+    # If for some reason languages = None, assume this also means 'all'
+    if isinstance(languages, str) and languages != 'all':
+        langs_list = [languages]
+    elif isinstance(languages, list) or isinstance(languages, tuple):
+        if languages[0] != 'all':
+            langs_list = languages
+
+    manifests = defaultdict(dict) 
+    for lang in langs_list:
+        corpus_dir_lang = corpus_dir / f'{lang}-{lang}'
+        output_dir_lang = output_dir / f'{lang}'
+        if corpus_dir_lang.is_dir():
+            manifests[lang] = prepare_single_mtedx_language(
+                corpus_dir_lang,
+                output_dir_lang,
+                language=lang,
+                num_jobs=num_jobs,
+            )
+    
+    return dict(manifests)
+
+
+###############################################################################
+# All remaining functions are just helper functions, mainly for text
+# normalization and parsing the vtt files that come with the mtedx corpus
+###############################################################################
+
+# Prepare data for a single language
+def prepare_single_mtedx_language(
+    corpus_dir: Pathlike,
+    output_dir: Optional[Pathlike] = None,
+    language: str = 'language',
```

Comment:
The default argument choice seems weird, if we treat it as an internal function maybe it’s better to remove the default value?

Reply:
Sure. There was one other point which I wanted to ask about. The dumped json files for arabic for instance print the unicode values using the \u00... format, rather than something readable, which might be of interest. If you use the ascii_only=False flag on json.dump then you can read the output, but maybe there was a reason for serializing with ascii only characters?

Reply:
Hmm theoretically we’re in the 21st century and everything should be ok if we just write Unicode in the manifests. Yeah… sure. 🤥

I’d suggest to allow writing Unicode by default and if somebody needs to turn it off, they can always manually encode the text values. Or we can add flags to customize this behavior later. WDYT?

Reply:
JSON RFC https://www.ietf.org/rfc/rfc4627.txt

`3.  Encoding

   JSON text SHALL be encoded in Unicode.  The default encoding is
   UTF-8. `

Reply:
I guess I don't really know what SHALL means here. At some point in the future, or just recommended? I think as long as there aren't cases where writing and then reading cause some corruption of the data. it's preferable to be able to read the manifest. The easiest change is just setting ensure_ascii=False in the save_to_json file and then always doing that, but I don't know if that is back-compatible or if that causes other problems.

Reply:
I agree with you. I think it is unlikely to break anything.

Reply:
Do you want that update in this PR, or in a separate commit? I guess we might want to change jsonl as well? I'm not as familiar with the other formats.

Reply:
Yeah, a follow-up PR is a good idea, let's change JSONL as well (and YAML too if they allow it -- it's only for didactic purposes so it'd be good to have it there too)